File: sample-java/src/main/java/com/github/charleslzq/samplejava/SampleApplication.java
package com.github.charleslzq.samplejava;

import android.app.Application;

import com.orhanobut.logger.AndroidLogAdapter;
import com.orhanobut.logger.Logger;

public class SampleApplication extends Application {
    private static String LOG_DIRECTORY = "/logs/java";

    @Override
    public void onCreate() {
        super.onCreate();
        System.setProperty("rx2.buffer-size", "20");

        Logger.addLogAdapter(new AndroidLogAdapter());

        Thread.setDefaultUncaughtExceptionHandler(new Thread.UncaughtExceptionHandler() {
            @Override
            public void uncaughtException(Thread t, Throwable e) {
                Logger.e(e, "Exception occur at %s", t.getName());
            }
        });
    }
}

File: sample/src/main/java/com/github/charleslzq/sample/SampleApplication.kt
package com.github.charleslzq.sample

import android.app.Application
import com.orhanobut.logger.AndroidLogAdapter
import com.orhanobut.logger.Logger


private const val LOG_DIRECTORY = "/logs"

class SampleApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        System.setProperty("rx2.buffer-size", "20")

        Logger.addLogAdapter(AndroidLogAdapter())

        Thread.setDefaultUncaughtExceptionHandler { thread, throwable ->
            Logger.e(throwable, "Exception occur at ${thread.name}")
        }
    }
}